src/App.vue

```diff
@@ -1,9 +1,55 @@
 <template>
   <div id="app">
-    <router-view />
+    <transition :name="transitionName" mode="out-in">
+      <router-view />
+    </transition>
   </div>
 </template>
 
+<script>
+export default {
+  data() {
+    return {
+      transitionName: 'forward',
+    }
+  },
+  watch: {
+    '$route' (to, from) {
+      const toDepth = to.path.split('/').length
+      const fromDepth = from.path.split('/').length
+      this.transitionName = toDepth < fromDepth ? 'back' : 'forward'
+    }
+  },
+}
+</script>
+
 <style>
 @import url("./assets/css/style.css");
+
```

Comment:
Doesn't app.vue gets loaded in everywhere?
If so, there is probably no need to put inline css but it can be placed to the global less styling?
If not, don't mind this comment :)

Reply:
Yes i guess we should have everything In less and compile one general style and scoped ones for each view.